libs/Einsums/LinearAlgebra/include/Einsums/LinearAlgebra/Bases/sum_square.hpp

```diff
@@ -0,0 +1,98 @@
+//----------------------------------------------------------------------------------------------
+// Copyright (c) The Einsums Developers. All rights reserved.
+// Licensed under the MIT License. See LICENSE.txt in the project root for license information.
+//----------------------------------------------------------------------------------------------
+
+#pragma once
+#include <Einsums/BLAS.hpp>
+#include <Einsums/Profile/LabeledSection.hpp>
+#include <Einsums/TensorImpl/TensorImpl.hpp>
+
+namespace einsums {
+namespace linear_algebra {
+namespace detail {
+
+template <typename T>
+void impl_sum_square_contiguous(einsums::detail::TensorImpl<T> const &a, RemoveComplexT<T> *scale, RemoveComplexT<T> *sumsq) {
+    if constexpr (blas::IsBlasableV<T>) {
+        blas::lassq(a.size(), a.data(), a.get_incx(), scale, sumsq);
+    } else {
+        T const     *a_data = a.data();
+        size_t const size = a.size(), incx = a.get_incx();
+        *scale = T{1.0};
+
+        EINSUMS_OMP_SIMD_PRAGMA(parallel for reduction(+: *sumsq))
```

Comment:
You can't use a pointer dereference here `*sumsq`. You'll have to have a local variable here and then set `*sumsq = local_variable`.